views/templates/hook/displayBlock.tpl

```diff
@@ -22,22 +22,51 @@
 *  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
 *  International Registered Trademark & Property of PrestaShop SA
 *}
-<div class="blockreas">
-    {foreach from=$blocks item=$block key=$key}
-        <div class="reass-item" style="{if $block['type_link'] !== $LINK_TYPE_NONE && !empty($block['link'])}cursor:pointer;{/if}"
-                {if ($block['type_link'] !== $LINK_TYPE_NONE) && !empty($block['link'])} onclick="window.open('{$block['link']}')"{/if}>
-            <div class="block-icon">
-                {if $block['icon'] != 'undefined'}
-                    {if $block['icon']}
-                        <img class="svg invisible" src="{$block['icon']}"">
-                    {elseif $block['custom_icon']}
-                        <img src="{$block['custom_icon']}">
+<div class="container">
+    <div class="blockreas">
+        {assign var=numCols value=$blocks|@count}
+        {assign var=numColsRemaining_md value=($numCols % 4)}
+        {assign var=numColsRemaining_sm value=($numCols % 2)}
+        <div class="row">
+        {foreach from=$blocks item=$block key=$key name=blocks}
+            {assign var=idxCol value=($smarty.foreach.blocks.index + 1)}
+            {assign var=sizeCol_md value=3}
+            {assign var=offsetCol_md value="offset-md-0"}
+            {assign var=sizeCol_sm value=6}
+            {assign var=offsetCol_sm value="offset-sm-0"}
+            {if $idxCol > ($numCols - $numColsRemaining_md)}
+                {if $numColsRemaining_md == 2}
+                    {if !$smarty.foreach.blocks.last}
+                        {assign var=offsetCol_md value="offset-md-3"}
                     {/if}
+                {else}
+                    {assign var=sizeCol_md value=(12 / $numColsRemaining_md)}
+                {/if}
+            {/if}
+            {if $idxCol > ($numCols - $numColsRemaining_sm)}
+                {if $numColsRemaining_md == 1}
+                    {assign var=offsetCol_sm value="offset-sm-3"}
+                {else}
+                    {assign var=sizeCol_sm value=(12 / $numColsRemaining_md)}
                 {/if}
+            {/if}
+            <div class="col-md-{$sizeCol_md} {$offsetCol_md} col-sm-{$sizeCol_sm} {$offsetCol_sm} col-xs-12"
+                {if $block['type_link'] !== $LINK_TYPE_NONE && !empty($block['link'])} style="cursor:pointer;"{/if}
```

Comment:
Duplicate condition on lines 54 and 55 ? 🤔

Reply:
:ok_hand:

Reply:
There's several of them by the way